.circleci/config.yml

```diff
@@ -10,61 +10,26 @@ jobs:
     steps:
       - checkout
 
-      # Factoring demo
-#      - run: &make-virtualenv
-#          command: |
-#            python -m virtualenv env
-#
-#      - run:
-#          command: |
-#            . env/bin/activate
-#            pip install -r factoring/requirements_tests.txt
-#
-#TODO: add caching back in, maybe use loop
+      #TODO: add caching once demos share the same requirements file and can live in the same venv
       - run:
+          name: install dependences and run tests
           command: |
-            demos=$(ls -d */ | grep -v "tests" | grep -v "circuit-fault-diagnosis" | sed 's/\///')
-            for demo in $demos
-            do
-                env="${demo}_env"
-                python -m virtualenv $env
-                . $env/bin/activate
-                pip install -r $demo/requirements_tests.txt
+            shopt -s extglob
+
+            for demo in !(tests|circuit-fault-diagnosis); do
+              if [ -d "$demo" ]; then
+                env=""$demo"_env"
```

Comment:
Esthetic remark: you can just wrap the whole expression with double quotes:
```
env="${demo}_env"
```
Or if variable separator is not a valid variable name character (like `_` above is), you don't have to use braces:
```
env="env_$demo"
```